share/completions/gradle.fish

```diff
@@ -37,3 +37,24 @@ complete -c gradle -l continuous -s t -d 'Enable continuous build'
 complete -c gradle -l no-search-upward -s u -d 'Don\'t search in parent folders for settings file'
 complete -c gradle -l version -s v -d 'Print version'
 complete -c gradle -l exclude-task -s x -x -d 'Specify task to be excluded from execution'
+
+# https://github.com/hanny24/gradle-fish/blob/master/gradle.load
+function __cache_or_get_gradle_completion
+  mkdir -p "/tmp/gradle_completion_cache_for_$USER"
+  set -l hashed_pwd (echo $PWD | md5sum | awk '{ print $1 }')
+  set -l gradle_cache_file "/tmp/gradle_completion_cache_for_$USER/$hashed_pwd"
+  if /usr/bin/test build.gradle -nt "$gradle_cache_file"; or not test -f "$gradle_cache_file"
```

Comment:
>I'm guessing that runs around the builtin fish test then?

Yes. It tells fish that only external commands should be tried - i.e. instead of running a function if it exists, or a builtin if it exists and only then an external command, it goes directly to the $PATH lookup stage.

It'll end up running "/usr/bin/test" if your test resides there, but on systems where /bin and /usr/bin are separate, it's usually (IIRC) "/bin/test" instead. So `/usr/bin/test` would fail, while `command test` would succeed.